pkg/stanza/operator/helper/time.go

```diff
@@ -47,12 +47,11 @@ type TimeParser struct {
 
 // Unmarshal starting from default settings
 func (t *TimeParser) Unmarshal(component *confmap.Conf) error {
-	cfg := NewTimeParser()
-	err := component.Unmarshal(&cfg, confmap.WithIgnoreUnused())
+	*t = NewTimeParser()
```

Comment:
Why do we need to mutate `t` if `Unmarshal` fails?

Reply:
oh alright I'll put that after.

Reply:
done